tests/admin_scripts/tests.py

```diff
@@ -2114,6 +2114,29 @@ def test_invalid_name(self):
                 )
                 self.assertFalse(os.path.exists(testproject_dir))
 
+    def test_invalid_target_name(self):
+        """startapp validates that target dir name is a valid Python identifier."""
+        for bad_target in ('invalid.dir_name', '7invalid_dir_name'):
```

Comment:
As this test runs several inner tests, you can use `unittest.subTest` to provide more useful output on each iteration.

https://docs.python.org/3/library/unittest.html#unittest.TestCase.subTest

Something like:

```py
for bad_target in ['invalid.dir_name', '7invalid_dir_name']:
    with self.subTest(bad_target):
        ...
```